test/unittest/conftest.py

```diff
@@ -117,247 +56,116 @@ def an_insert(float_tensors_32):
+    ]
+    for identifier, weight_shape, encoder in params:
+        data_layer.add(
+            TorchModel(
+                object=torch.nn.Linear(*weight_shape),
+                identifier=identifier,
+                encoder=encoder,
             )
         )
-        vi = VectorIndex(
-            identifier=identifier,
-            indexing_listener='linear_a/x',
-            compatible_listener='linear_a/z',
-            **kwargs,
-        )
-        db.add(vi)
-        return vi
 
-    return _factory
 
-
-@pytest.fixture()
-def with_vector_index(random_data, vector_index_factory):
-    vector_index_factory(random_data, 'test_vector_search')
-    yield random_data
+def add_vector_index(
+    data_layer: Datalayer, collection_name='documents', identifier='test_vector_search'
+):
+    # TODO: Support configurable key and model
+    data_layer.add(
+        Listener(
+            select=Collection(collection_name).find(),
+            key='x',
+            model='linear_a',
+        )
+    )
+    data_layer.add(
+        Listener(
+            select=Collection(collection_name).find(),
+            key='z',
+            model='linear_a',
+        )
+    )
+    vi = VectorIndex(
+        identifier=identifier,
+        indexing_listener='linear_a/x',
+        compatible_listener='linear_a/z',
+    )
+    data_layer.add(vi)
 
 
-@pytest.fixture()
-def si_validation(random_data):
-    d = Dataset(
-        identifier='my_valid',
-        select=Collection('documents').find({'_fold': 'valid'}),
-        sample_size=100,
-    )
-    random_data.add(d)
-    yield random_data
+@pytest.fixture(scope='session')
+def image_url():
+    path = Path(__file__).parent.parent / 'material' / 'data' / '1x1.png'
+    return f'file://{path}'
 
 
-@pytest.fixture()
-def float_tensors_32(empty):
-    empty.add(tensor(torch.float, shape=(32,)))
-    yield empty
-    empty.remove('encoder', 'torch.float32[32]', force=True)
+@pytest.fixture(scope='session')
+def global_identifier_of_vector_index() -> str:
+    return 'global_identifier_of_vector_index'
 
 
-@pytest.fixture()
-def float_tensors_16(empty):
-    empty.add(tensor(torch.float, shape=(16,)))
-    yield empty
-    empty.remove('encoder', 'torch.float32[16]', force=True)
+def setup_data_layer(data_layer, **kwargs):
+    # TODO: support more parameters to control the setup
+    add_encoders(data_layer)
+    n_data = kwargs.get('n_data', n_data_points)
+    add_random_data(data_layer, number_data_points=n_data)
+    if kwargs.get('add_models', True):
+        add_models(data_layer)
+    if kwargs.get('add_vector_index', True):
+        add_vector_index(data_layer)
 
 
-@pytest.fixture()
-def float_tensors_8(empty):
-    empty.add(tensor(torch.float, shape=(8,)))
-    yield empty
-    empty.remove('encoder', 'torch.float32[8]', force=True)
+@pytest.fixture(scope='session')
+def data_layer() -> Datalayer:
+    _data_layer = build_datalayer(CFG, data_backend=MONGOMOCK_URI)
```

Comment:
private var in function closure are not required we can use simple data_layer 
:)

Reply:
Yes, done